pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -731,8 +732,11 @@ class Timestamp(_Timestamp):
         if ts.value == NPY_NAT:
             return NaT
 
-        if is_string_object(freq):
+        if is_string_object(freq) or isinstance(freq, _BaseOffset):
```

Comment:
don't do this as could have an adverse perf impact (this is a python object). use ``is_offset_object``